### PR TITLE
LoggingKata Push

### PR DESCRIPTION
**Pull request by HunterAW**
This is the initial push. All steps have not yet been completed and lots of refactoring needs to be done.

### CODE REVIEW

File: LoggingKata.Test/ParserTest.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
```
Comment (codytwinton) [inline LoggingKata.Test/ParserTest.cs]: This directive seems unnecessary

File: LoggingKata.Test/ParserTest.cs [hunk 2]
```diff
@@ -10,10 +11,30 @@ namespace LoggingKata.Test
     [TestFixture]
     public class TacoParserTestFixture
     {
+        [Test]
+        public void ShouldReturnNullForEmptyString()
```
Comment (codytwinton) [inline LoggingKata.Test/ParserTest.cs]: You should also write tests for:

- `null` string should return `null`
- `string` with lat and long, but no description returns valid `ITrackable`
- `string` without both lat and long should return `null`

File: LoggingKata/Program.cs [hunk 3]
```diff
@@ -31,6 +32,33 @@ static void Main(string[] args)
             //TODO:  Find the two TacoBells in Alabama that are the furthurest from one another.
             //HINT:  You'll need two nested forloops
 
+
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Missing implementation

File: LoggingKata/TacoParser.cs [hunk 4]
```diff
@@ -20,7 +23,46 @@ public TacoParser()
         public ITrackable Parse(string line)
         {
             //DO not fail if one record parsing fails, return null
-            return null; //TODO Implement
+            //TODO Implement
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: This implementation will fail for all the cases you still require tests for

File: LoggingKata/Program.cs [hunk 5]
```diff
@@ -5,17 +5,18 @@
...
     {
-        //Why do you think we use ILog?
         private static readonly ILog Logger =
             LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
 
         static void Main(string[] args)
         {
+            var myFile = Environment.CurrentDirectory + "\\Taco_Bell-US-AssemblyLoadEventArgs-Alabama.csv";
 
             if (args.Length == 0)
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Remove this, since this will fail without any arguments

File: LoggingKata/Program.cs [hunk 6]
```diff
@@ -24,41 +25,37 @@ static void Main(string[] args)
                 return;
             }
 
+            if (args.Length == 1)
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: This will fail without arguments and we don't need arguments.

File: LoggingKata/Program.cs [hunk 7]
```diff
@@ -24,41 +25,37 @@ static void Main(string[] args)
...
+            {
+                Console.WriteLine("You must provide a filename as an argument");
+                Logger.Fatal("Cannot import without filename specified as an argument");
+                return;
+            }
+
             Logger.Info("Log initialized");
-            var lines = File.ReadAllLines(args[0]);
+            var lines = File.ReadAllLines(myFile);
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Add some logs about the lines. If lines.Length == 0 might be one.

File: LoggingKata/Program.cs [hunk 8]
```diff
@@ -24,41 +25,37 @@ static void Main(string[] args)
...
-            // Do a loop for your locations to grab each location as the origin (perhaps: `locA`)
-            // Create a new Coordinate with your locA's lat and long
-
-            // Now, do another loop on the locations with the scope of your first loop, so you can grab the "destination" location (perhaps: `locB`)
-            // Create a new Coordinate with your locB's lat and long
-            // Now, compare the two using `GeoCalculator.GetDistance(origin, destination)`, which returns a double
-            // If the distance is greater than the currently saved distance, update the distance and the two `ITrackable` variables you set above
-
-            // Once you've looped through everything, you've found the two Taco Bells furthest away from each other.
+            /*
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Needs work.

File: LoggingKata/TacoParser.cs [hunk 9]
```diff
@@ -1,26 +1,46 @@
...
+
+                return new TacoBell
+                {
+                    Name = cells.Length > 2 ? cells[2] : null,
+                    Location = new Point(lat, lon)
+                };
+            }
+            catch (Exception e)
+            
+            {
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: You should log the Exception

File: LoggingKata/TacoParser.cs [hunk 10]
```diff
@@ -1,26 +1,46 @@
...
         {
-            //DO not fail if one record parsing fails, return null
-            return null; //TODO Implement
+            if (string.IsNullOrEmpty(line)) { return null; }
+
+            var cells = line.Split(',');
+
+            if (cells.Length < 2)
+            {
+                return null;
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: You should log something here.

File: LoggingKata/Program.cs [hunk 11]
```diff
@@ -5,32 +5,67 @@
...
+                Console.ReadLine();
+                Logger.Fatal("Cannot import without filename specified as an argument");
+                return;
+            }
+
+            if (lines.Length == 1)
             {
-                Console.WriteLine("You must provide a filename as an argument");
+                Console.WriteLine("You must provide a filename as an argument args.Length == 1");
+                Console.ReadLine();
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Not sure what the ReadLine here is for.

**Comment by HunterAW** [comment]
I must have been on autopilot in CMD I hadn't meant to push that version.
Though the one I just pushed doesn't contain all the logging and testing
that it needs it is a better version than the one you looked at.

**Pull request closed without merging**